packages/loot-core/src/server/db/index.js

```diff
@@ -565,14 +565,31 @@ export async function insertAccount(account) {
     [account.offbudget != null ? account.offbudget : 0]
   );
 
+  if (accounts.some(existingAccount => existingAccount.name === account.name)) {
+    return 'Account name already exists';
+  }
+
   // Don't pass a target in, it will default to appending at the end
   let { sort_order } = shoveSortOrders(accounts);
 
   account = accountModel.validate({ ...account, sort_order });
   return insertWithUUID('accounts', account);
 }
 
-export function updateAccount(account) {
+export async function updateAccount(account) {
+  const accounts = await getAccounts();
+
+  if (
+    accounts.some(
+      existingAccount =>
+        existingAccount.name === account.name &&
+        // If it's the current account name, then there's no issue.
+        existingAccount.id !== account.id
+    )
+  ) {
+    return 'Account name already exists';
```

Comment:
I’m not a fan of overloading the return value here. Can you refactor this function to return either `{ error: 'already-exists' }` or `{ id }`? That way the caller can check `if (error) return { error }` to propagate the error.